src/App.tsx

```diff
@@ -1,26 +1,332 @@
+    deleteTodo(id)
+      .then(() => {
+        setTodos(currentTodos => currentTodos.filter(todo => todo.id !== id));
+        setdeletingId(null);
+        focusInput();
+      })
+      .catch(() => {
+        setErrorMessage('Unable to delete a todo');
+        setdeletingId(null);
+        focusInput();
+      });
+  }
+
+  function addTodo(): Promise<void> {
+    setErrorMessage('');
+
+    if (!value.trim()) {
+      setErrorMessage('Title should not be empty');
+
+      return Promise.reject();
+    }
+
+    setTempTodo({ id: 0, completed: false, title: value, userId: USER_ID });
+
+    return createTodos({ userId: USER_ID, title: value, completed: false })
+      .then(newTodo => {
+        setTodos(currentTodos => [...currentTodos, newTodo]);
+        setTempTodo(null);
+        focusInput();
+      })
+      .catch(error => {
+        setErrorMessage('Unable to add a todo');
+        setTempTodo(null);
+        focusInput();
+        throw error;
+      });
+  }
+
+  const handleAddingTodo = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setErrorMessage('');
+    setValue(e.target.value);
+  };
+
+  const handleSubmit = (e: React.ChangeEvent<HTMLFormElement>) => {
+    setIsSubmitting(true);
+    e.preventDefault();
+    if (!value.trim()) {
+      setErrorMessage('Title should not be empty');
+      setIsSubmitting(false);
+      inputRef.current?.focus();
+
+      return;
+    }
+
+    addTodo()
+      .then(() => setValue(''))
+      .finally(() => setIsSubmitting(false));
+  };
+
+  const handleFiltering = (type: FilterTypes) => {
+    setFilteringType(type);
+  };
+
+  const clearCompletedTodo = () => {
+    completedTodos.map(todo => deleteSingleTodo(todo.id));
+    focusInput();
+  };
+
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">
-          React Todo App - Load Todos
-        </a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+          <button
+            type="button"
+            className={classNames('todoapp__toggle-all', {
+              active: allCompletedTodos,
+            })}
+            data-cy="ToggleAllButton"
+          />
+
+          {/* Add a todo on form submit */}
+          <form onSubmit={handleSubmit}>
+            <input
+              ref={inputRef}
+              disabled={isSubmitting}
+              data-cy="NewTodoField"
+              type="text"
+              value={value}
+              onChange={handleAddingTodo}
+              className="todoapp__new-todo"
+              placeholder="What needs to be done?"
+            />
+          </form>
+        </header>
```

Comment:
Splitting the App into multiple components helps with readability and maintainability. `Header` could be a separate component